compiler-core/src/language_server/code_action.rs

```diff
@@ -276,3 +276,89 @@ impl<'a> RedundantTupleInCaseSubject<'a> {
         }
     }
 }
+
+pub struct AssertResultToCase<'a> {
+    module: &'a Module,
+    params: &'a CodeActionParams,
+    actions: Vec<CodeAction>,
+    line_numbers: LineNumbers,
+}
+
+impl<'ast> ast::visit::Visit<'ast> for AssertResultToCase<'_> {
+    fn visit_typed_assignment(&mut self, assignment: &'ast ast::TypedAssignment) {
+        let range = src_span_to_lsp_range(assignment.location, &self.line_numbers);
+
+        if !overlaps(range, self.params.range) {
+            return;
+        }
+
+        if !matches!(assignment.kind, AssignmentKind::Assert { .. }) {
+            return;
+        };
+
+        let Pattern::Constructor {
+            name,
+            arguments,
+            type_,
+            ..
+        } = &assignment.pattern
+        else {
+            return;
+        };
+
+        if type_.is_result() && name == "Ok" {
```

Comment:
I think we would like this to work for any pattern rather than just `Ok`. Do you think you'd be up for implementing that?

You could get the span of the pattern and copy the string contents of that range to make the edit.

We could either for now we could put a `todo` on the right hand side of the `->`, or do the work to figure out all the variables that get bound and then make a tuple of them. Might not be too tricky using another visitor applied to the pattern.

Reply:
Yep, I can do that. Can you give an example of how it would work with capturing all the variables? I don't completely get what you mean by that.

Reply:
Oh I think I see. You just mean for the pattern that's asserted? Not for the other possible cases

Reply:
Some examples:
---
```gleam
let assert Ok(x) = y
```
would need to generate
```gleam
let x = case y {
  Ok(x) -> x
  _ -> panic
}
```
---
```gleam
let assert [x, 1, y] = z
```
would need to generate
```gleam
let #(x, y) = case z {
  [x, 1, y] -> #(x, y)
  _ -> panic
}
```

Reply:
Right. So we would just do the `_` case for other options, no inference of other patterns? That makes it much easier to implement. Maybe once #2957 is complete, we can add that functionality.